src/main/java/zmq/ZMQ.java

```diff
@@ -822,14 +822,23 @@ public Selector get()
             return selector;
         }
 
+        void close()
+        {
+            if (selector != null) {
+                try {
+                    selector.close();
+                    selector = null;
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
         @Override
         public void finalize()
         {
```

Comment:
This PollSelector class is not part of libzmq, as far as I can see. ATM, it was a workaround to comply with the original library, so this code may be refactored, or rather its use enforced to provide a correct behaviour.